snapcraft/yaml.py

```diff
@@ -320,30 +323,95 @@ def _build_env(root):
         '-L{0}/usr/lib/{1}',
         '$LDFLAGS'
     ]).format(root, arch_triplet) + '"')
-    env.append('PKG_CONFIG_PATH=' + ':'.join([
-        '{0}/lib/pkgconfig',
-        '{0}/lib/{1}/pkgconfig',
-        '{0}/usr/lib/pkgconfig',
-        '{0}/usr/lib/{1}/pkgconfig',
-        '{0}/usr/share/pkgconfig',
-        '{0}/usr/local/lib/pkgconfig',
-        '{0}/usr/local/lib/{1}/pkgconfig',
-        '{0}/usr/local/share/pkgconfig',
-        '$PKG_CONFIG_PATH'
-    ]).format(root, arch_triplet))
 
     return env
 
 
 def _build_env_for_stage(stagedir):
     env = _build_env(stagedir)
-
     env.append('PERL5LIB={0}/usr/share/perl5/'.format(stagedir))
-    env.append('PKG_CONFIG_SYSROOT_DIR={0}'.format(stagedir))
 
     return env
 
 
+_PKG_CONFIG_TEMPLATE = """#!/usr/bin/env python3
+
+import os
+import sys
+from subprocess import check_call
+
+real_env = os.environ.copy()
+
+
+def run_pkg_config(args, env):
+    check_call(['/usr/bin/pkg-config'] + args, env=env)
+
+
+def get_pkg_env_for(basedir):
+    current_env = real_env.copy()
+    env = {{}}
+    env['PKG_CONFIG_PATH'] = ':'.join([
+        '{{basedir}}/lib/pkgconfig',
+        '{{basedir}}/lib/{arch}/pkgconfig',
+        '{{basedir}}/usr/lib/pkgconfig',
+        '{{basedir}}/usr/lib/{arch}/pkgconfig',
+        '{{basedir}}/usr/share/pkgconfig',
+        '{{basedir}}/usr/local/lib/pkgconfig',
+        '{{basedir}}/usr/local/lib/{arch}/pkgconfig',
+        '{{basedir}}/usr/local/share/pkgconfig']).format(basedir=basedir)
+    if 'PKG_CONFIG_PATH' in current_env:
+        env['PKG_CONFIG_PATH'] = '{{}}:{{}}'.format(
+            env['PKG_CONFIG_PATH'], current_env['PKG_CONFIG_PATH'])
+    env['PKG_CONFIG_SYSROOT_DIR'] = {{basedir}}
+
+    return current_env.update(env)
+
+
+def modules_exist(modules, env):
+    try:
+        check_call(['/usr/bin/pkg-config', '--exists'] + modules, env=env)
+        return True
+    except:
```

Comment:
except subprocess.CalledProcessError.
We don't want to ignore any other exceptions.